addons/account/models/account_move.py

```diff
@@ -1749,7 +1749,7 @@ def write(self, vals):
             self.mapped('line_ids')._check_tax_lock_date()
 
         if ('state' in vals and vals.get('state') == 'posted'):
-            for move in self.filtered(lambda m: m.restrict_mode_hash_table and not(m.secure_sequence_number or m.inalterable_hash)):
+            for move in self.filtered(lambda m: m.restrict_mode_hash_table and not(m.secure_sequence_number or m.inalterable_hash)).sorted(lambda m: (m.date, m.ref or '', m.id)):
```

Comment:
Is there a need to do 2 separate commit for this bug fix? Or are the both of them related? If they are related to the same problem you can just have one commit with the details in the commit message. Also a bit more explanation in the commit message would be nice to explain why this fix is needed because I have no idea why we have to sort it by date then by ref and finally by id. What was the problem before the fix and why did we fix it like that should probably be explained so that later on we can understand easily by looking up the commit message :)

Reply:
We could merged the 2 commit into 1.
Yep, I'll do it for commit message :)